pkg/operator/configobservation/network/observe_network.go

```diff
@@ -10,57 +10,93 @@ import (
-		}
-		if err := unstructured.SetNestedStringSlice(previouslyObservedConfig, currentRestrictedCIDRBs, restrictedCIDRsPath...); err != nil {
-			errs = append(errs, err)
-		}
+	admissionControllerConfig := unstructured.Unstructured{}
+	prev, ok, err := unstructured.NestedMap(existingConfig, configPath...)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	if ok && err != nil {
+		// If we ever bump the API version, we'll have to do conversion here.
+		admissionControllerConfig.SetUnstructuredContent(prev)
 	}
+	admissionControllerConfig.SetAPIVersion("network.openshift.io/v1")
+	admissionControllerConfig.SetKind("RestrictedEndpointsAdmissionConfig")
 
-	observedConfig := map[string]interface{}{}
 	clusterCIDRs, err := network.GetClusterCIDRs(listers.NetworkLister, recorder)
 	if err != nil {
 		errs = append(errs, err)
-		return previouslyObservedConfig, errs
 	}
 	serviceCIDR, err := network.GetServiceCIDR(listers.NetworkLister, recorder)
 	if err != nil {
 		errs = append(errs, err)
+	}
+
+	// If we weren't able to retrieve cidrs, then return the previous configuration
+	if len(errs) > 0 {
+		previouslyObservedConfig := map[string]interface{}{}
+		unstructured.SetNestedMap(previouslyObservedConfig, admissionControllerConfig.Object, configPath...)
 		return previouslyObservedConfig, errs
 	}
 
 	// set observed values
-	//  admissionPluginConfig:
-	//    network.openshift.io/RestrictedEndpointsAdmission:
-	//	  configuration:
-	//	    restrictedCIDRs:
-	//	    - 10.3.0.0/16 # ServiceCIDR
-	//	    - 10.2.0.0/16 # ClusterCIDR
-	//  servicesSubnet: 10.3.0.0/16
+	//  admission:
+	//    pluginConfig:
+	//      network.openshift.io/RestrictedEndpointsAdmission:
+	//        version: network.openshift.io/v1
+	//        kind: RestrictedEndpointsAdmission
+	//        configuration:
+	//          restrictedCIDRs:
+	//            - 10.3.0.0/16 # ServiceCIDR
+	//            - 10.2.0.0/16 # ClusterCIDR
 	restrictedCIDRs := clusterCIDRs
 	if len(serviceCIDR) > 0 {
 		restrictedCIDRs = append(restrictedCIDRs, serviceCIDR)
 	}
-	if len(restrictedCIDRs) > 0 {
-		if err := unstructured.SetNestedStringSlice(observedConfig, restrictedCIDRs, restrictedCIDRsPath...); err != nil {
-			errs = append(errs, err)
-		}
+
+	if err := unstructured.SetNestedStringSlice(admissionControllerConfig.Object, restrictedCIDRs, "restrictedCIDRs"); err != nil {
+		errs = append(errs, err)
 	}
-	if len(serviceCIDR) > 0 {
-		if err := unstructured.SetNestedField(observedConfig, serviceCIDR, "servicesSubnet"); err != nil {
-			errs = append(errs, err)
+
+	observedConfig := map[string]interface{}{}
+	unstructured.SetNestedMap(observedConfig, admissionControllerConfig.Object, configPath...)
+
+	return observedConfig, errs
+}
+
+// ObserveServicesSubnet watches the network configuration and generates the
+// servicesSubnet
+func ObserveServicesSubnet(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
```

Comment:
This appears to only be used from TestObserveServicesSubnet()?

Reply:
good catch, fixed.